env0/resource_aws_eks_credentials.go

```diff
@@ -0,0 +1,64 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsEksCredentials() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAwsEksCredentialsCreate,
+		ReadContext:   resourceCredentialsRead(AWS_EKS_TYPE),
+		DeleteContext: resourceCredentialsDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceCredentialsImport(AWS_EKS_TYPE)},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name for the credentials",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"cluster_name": {
+				Type:        schema.TypeString,
+				Description: "eks cluster name",
+				Required:    true,
+				ForceNew:    true,
```

Comment:
You can update properties (other than name) in place, so I think there's no need to force a new resource
This comment is relevant to all other credential types added in this PR

Reply:
added the update/patch option. (see latest commit for specific changes).